proposals/nnnn-structured-concurrency.md

```diff
@@ -168,9 +174,49 @@ The execution of a task can be seen as a succession of periods where the task wa
+
+Swift provides a default executor implementation, but both actor classes and global actors can suppress this and provide their own implementation.
+
+Generally end-users need not interact with executors directly, but rather use them implicitly by invoking actors and functions which happen to use executors to perform the invoked asynchronous functions.
+
+### Task priorities
+Any task is associated with a specific `Task.Priority`.
+
+Task priority may inform decisions an `Executor` makes about how and when to schedule tasks submitted to it. An executor may utilize priority information to attempt running higher priority tasks first, and then continuing to serve lower priority tasks.
+
+The exact semantics of how priority is treated are left up to each platform and specific `Executor` implementation.
+
+Child tasks automatically inherit their parent task's priority. Detached tasks do not inherit priority (or any other information) because they semantically do not have a parent task.
+
+```swift
+extension Task {
+  public static func currentPriority() async -> Priority { ... }
+
+  public struct Priority: Comparable {
+    public static let `default`: Task.Priority
+    /* ... */
+  }
+}
+```
+
+> **TODO**: Define the details of task priority; It is likely to be a concept similar to Darwin Dispatch's QoS; bearing in mind that priority is not as much of a thing on other platforms (i.e. server side Linux systems).
 
-An executor is called exclusive if the partial tasks submitted to it will never be run concurrently.  (Specifically, the partial tasks must be totally ordered by the happens-before relationship: given any two tasks that were submitted and run, the end of one must happen-before the beginning of the other.)  Executors are not required to run partial tasks in the order they were submitted; in fact, they should generally honor task priority over submission order.
+One of the ways to declare a priority level for a task is to pass it to `Task.runDetached(priority:operation:)` when starting a top-level task. All tasks started from within this task will inherit this task's priority since they would be its child tasks. 
+
+This means that, semantically, the "UI Thread" can be represented as a top-level *UI Task* which was started as detached with the `.ui` priority, and all other tasks which need to run on as children of the UI task, will inherit its priority. In practice this will likely be reflected by a global `UIActor` (see [global actors](nnnn-actors.md) in the actors proposal) which is designated to run on an UI thread assigned `Executor` which sets tasks it uses to use the UI priority, thus handling propagation of priority from tasks started from the UI actor itself.
+
+#### Priority Escalation
```

Comment:
I'm not sure I'm using the same phrase the way as @rjmccall has in mind -- is this the use of the phrase correct (in the section below)?

In wording it seems we mostly use the phrase "elevated" rather than escalated, but I assumed it's the same thing -- sanity checking if it is :)